.github/workflows/python-wheels.yml

```diff
@@ -68,24 +49,174 @@ jobs:
+    with:
+      os: '["ubuntu-22.04"]'
+      arch: "ppc64le"
+      test: "true"
+    secrets: inherit
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+  build_Linux_s390x:
+    name: Build Linux_s390x
+    uses: AcademySoftwareFoundation/openexr/.github/workflows/python-wheels-emulated.yml@main
+    with:
+      os: '["ubuntu-22.04"]'
+      arch: "s390x"
+      test: "true"
+    secrets: inherit
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+  build_MacOS_universal2:
+    name: Build MacOS_universal2
+    uses: AcademySoftwareFoundation/openexr/.github/workflows/python-wheels-emulated.yml@main
+    with:
+      os: '["macOS-latest"]'
+      arch: "universal2"
+      test: "true"
+    secrets: inherit
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+  publish:
+    runs-on: ${{ matrix.os }}
+    needs: [build_wheels, build_Linux_x86_64, build_Windows_AMD64, build_MacOS_x86_64, build_Linux_i686, build_Linux_aarch64, build_Linux_ppc64le, build_Linux_s390x, build_MacOS_universal2]
+    strategy:
+        matrix:
+            os: [ubuntu-22.04]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download sdist
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-sdist
+          path: dist/
+
+      - name: Download Linux_x86_64
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Linux_x86_64
+          path: dist/
+
+      - name: Download Windows_AMD64
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Windows_AMD64
+          path: dist/
+
+      - name: Download MacOS_x86_64
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-MacOS_x86_64
+          path: dist/
+
+      - name: Download Linux_i686
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Linux_i686
+          path: dist/
+
+      - name: Download Linux_aarch64
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Linux_aarch64
+          path: dist/
+
+      - name: Download Linux_ppc64le
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Linux_ppc64le
+          path: dist/
+
+      - name: Download Linux_s390x
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-Linux_s390x
+          path: dist/
+
+      - name: Download MacOS_universal2
+        uses: actions/download-artifact@v3
+        with:
+          name: python-wheels-MacOS_universal2
+          path: dist/
+
+      - name: Publish package to TestPyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
         with:
-          name: "Python wheels"
-          path: ./wheelhouse/*.whl
+          password: ${{ secrets.OPENEXR_PYPI_API }}
```

Comment:
You could use the new trusted publisher system instead of an API key: https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/#using-trusted-publishing-with-github-actions